src/hugr/typecheck.rs

```diff
@@ -79,8 +75,10 @@ fn check_valid_width(width: HugrIntWidthStore) -> Result<(), ConstTypeError> {
 
 /// Typecheck a constant value
 pub fn typecheck_const(typ: &ClassicType, val: &ConstValue) -> Result<(), ConstTypeError> {
-    match (typ, val) {
-        (ClassicType::Int(exp_width), ConstValue::Int { value, width }) => {
+    let failure = Err(ConstTypeError::TypeMismatch(typ.clone(), val.const_type()));
+    match typ {
```

Comment:
I think it would be neater to keep matching on the pair `(typ, val)`, rather than starting every branch with a `let ... else ...`

Reply:
I think falling through to returning `failure` is also neater than returning it in the branches

Reply:
Ok, there was a goal besides neatness (that it's easy to miss a case when you match on a pair), but never mind - I did write `[subjective]` in the title of that commit :-) :-)

I've removed that commit so now am just dropping `Failure` in favour of only TypeMismatch, how's that?